### PR TITLE
Add example notebook for the `jrc-gsw` collection

### PR DESCRIPTION
**Pull request by lossyrob**

### CODE REVIEW

File: datasets/jrc-gsw/jrc-gsw-example.ipynb [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cmaps = {}\n",
+    "for asset_key in cog_assets:\n",
+    "    asset = item.assets[asset_key]\n",
+    "    if asset_key in cog_assets:        \n",
```
Comment (TomAugspurger) [inline datasets/jrc-gsw/jrc-gsw-example.ipynb:164]: This `if` condition could be removed since it should always be true?

File: datasets/jrc-gsw/jrc-gsw-example.ipynb [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Returned 1 Items\n"
+     ]
+    }
+   ],
+   "source": [
+    "catalog = Client.open(\"https://planetarycomputer-staging.microsoft.com/api/stac/v1\")\n",
```
Comment (TomAugspurger) [inline datasets/jrc-gsw/jrc-gsw-example.ipynb:115]: Just a note to switch to prod when we're ready. That can be done as a followup.

**Review by TomAugspurger — COMMENTED** [review]
Up to you, but stackstack might still be worth using for the multi-asset handling:

```python
import stackstac
import rasterio.features

bbox = rasterio.features.bounds(area_of_interest)

ds = stackstac.stack(pc.sign(item), bounds_latlon=bbox)
ds

ds.squeeze().plot.imshow(col="band", col_wrap=3);
```

In this case, it may not be worth it since you need to iterate over the assets to get the colormap from the COGs anyway.

IIUC, colormaps can be baked into COGs in some sort of standard way. I wonder if it'd be possible to expose that via STAC, then stackstac could load that into the metadata of the xarray DataArray without needing to open the COG. And then we could teach xarray how to use those as the default colormap for an image.

**Comment by lossyrob** [comment]
>  stackstack might still be worth using for the multi-asset handling:

I had tried to use stacstack with the cmap handling a la the [io_lulc](https://github.com/microsoft/PlanetaryComputerExamples/blob/main/datasets/io-lulc/io-lulc-example.ipynb) notebook, but couldn't get the rendering to work right. Let me try again since I bailed on it pretty early, and either use it or come back with more specifics on how it wasn't working.

> IIUC, colormaps can be baked into COGs in some sort of standard way. I wonder if it'd be possible to expose that via STAC, then stackstac could load that into the metadata of the xarray DataArray without needing to open the COG

Related: https://github.com/stac-extensions/raster/issues/17

File: datasets/jrc-gsw/jrc-gsw-example.ipynb (hunk 1 above)
Comment (lossyrob) [reply datasets/jrc-gsw/jrc-gsw-example.ipynb]: yep, leftover from a different way of looping... fixed

**Comment by lossyrob** [comment]
I was able to plot the 6 bands with the example you provided, but since they all need their own colormap, I tried to pull a specific asset - but got stuck on:

```
bbox = rasterio.features.bounds(area_of_interest)

ds = stackstac.stack(pc.sign(item), bounds_latlon=bbox, assets=["occurrence"])
ds.squeeze().plot.imshow(col="band", cmap=cmaps["occurrence"])
```

resulting in 

```
...
/srv/conda/envs/notebook/lib/python3.8/site-packages/xarray/core/variable.py in __init__(self, dims, data, attrs, encoding, fastpath)
   2569         super().__init__(dims, data, attrs, encoding, fastpath)
   2570         if self.ndim != 1:
-> 2571             raise ValueError(f"{type(self).__name__} objects must be 1-dimensional")
   2572 
   2573         # Unlike in Variable, always eagerly load values into memory

ValueError: IndexVariable objects must be 1-dimensional
```

Surely this is me not knowing how to properly use stacstack, but for now I think moving forward with this is OK. Happy to pair on a stacstack solution or have this fixed up in a follow up.
